### PR TITLE
Decode JWT token fixes

### PR DESCRIPTION
**Pull request by orenbm**
This PR fixes 2 issues we had when decoding JWT tokens:
1. When we rescue a `JWT::DecodeError` we inspect the error to log it and to re-raise it but we forgot to mark the error in a variable.
2. We didn't rescue `Errno::ETIMEDOUT` when trying to discover the Oauth provider so we didn't raise a 504 Gateway Timeout error properly.

### CODE REVIEW

File: app/domain/authentication/o_auth/discover_identity_provider.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def discover_provider
         @discovered_provider = @open_id_discovery_service.discover!(@provider_uri).tap do
           @logger.debug(Log::IdentityProviderDiscoverySuccess.new)
         end
-      rescue HTTPClient::ConnectTimeoutError => e
+      rescue HTTPClient::ConnectTimeoutError, Errno::ETIMEDOUT => e
```
Comment (sigalsax) [inline app/domain/authentication/o_auth/discover_identity_provider.rb]: notice that also here, https://github.com/lostisland/faraday/issues/718
they use ::HTTPClient::TimeoutError, Errno::ETIMEDOUT

Seems that this fixes the issue but I would like to understand the difference between these 2.

Comment (sigalsax) [inline app/domain/authentication/o_auth/discover_identity_provider.rb]: Do we have a test for these two?

Comment (orenbm) [reply app/domain/authentication/o_auth/discover_identity_provider.rb]: We don't have tests for 504 as we had problems with this in the past. I will create an issue for this.

Comment (sigalsax) [reply app/domain/authentication/o_auth/discover_identity_provider.rb]: As discussed in person we need to investigate this issue further and design a test to mock a 504 GatewayTimeout and not a 502 BadGateway response

**Pull request merged**